### DISCUSSION TITLE
Low volume, BT does not last a reboot.

### DISCUSSION THREAD

**Discussion by phreatic62** [rern/rAudio#896] [Bug Reports]
HI I was lookng at another thread about low volume after an update and the solution was to go into 'player'. When I do this I get the error below, which goes away if I 'forget' my BT speaker.
I've also noticed that Bluetooth being turned on does not last through a reboot.

<img src="https://user-images.githubusercontent.com/61110813/163706753-82017cab-dcf2-49dc-8053-aa25462eed78.png" width=300>

THis appears to be since upgrading from rAudio-1-64bit-20211203 on a reapberry Pi 3

**Comment by rern** [comment]
Try:
`sed -i 's|amixer.*2|cat $dirshm/btclient 2> /dev/null|' /srv/http/bash/player-data.sh`

_(Edited: syntax correction)_

**Reply by phreatic62** [reply]
Hi Rern,

Thanks for the quick reply. Adding the line you gave now means that I can see the 'player' screen.
The volume is still low, but I get a speaker symbol to the right of the output device. When I use that the voloume changes and the volume on the main screen changes accordingly.
The volume is still quiet and BT being tuned on in the 'System' pages still does not survive a reboot.

Kind regards

**Comment by ceejayemm** [comment]
@rern 

Nearly same issue as @phreatic62 above,

RPi 3A, Newly flashed image i202204151 (no further updates), SoundCore 2 BT Speakers  (been using this combo for a while, no major problems in past)

After flashing image to new card, had same error as OP in post 1.  Edited player-data.sh file as advised by rern above. System/Network and Features pages Ok but Player page now gives black page with flashing rAudio logo.  If I click in the top left hand corner of the Player browser window I get:

{
  "page": "player",
  "devices": [
    {
      "aplayname": "bcm2835 Headphones",
      "card": 0,
      "device": 0,
      "dop": 0,
      "hw": "hw:0,0",
      "hwmixer": "Headphone",
      "mixers": 1,
      "mixerdevices": [
        "Headphone"
      ],
      "mixermanual": false,
      "mixertype": "none",
      "name": "On-board Headphones"
    }
  ],
  "active": true,
  "asoundcard": 0,
  "audioaplayname": "bcm2835 Headphones",
  "audiooutput": "On-board Headphones",
  "autoupdate": false,
  "btaplayname": "SoundCore 2 - A2DP",
  "buffer": false,
  "bufferconf": 4096,
  "bufferoutput": false,
  "bufferoutputconf": 16384,
  "counts": {
    "webradio": 4
  },
  "crossfade": false,
  "crossfadeconf": 1,
  "custom": false,
  "camilladsp": false,
  "equalizer": false,
  "ffmpeg": true,
  "normalization": false,
  "replaygain": false,
  "replaygainconf": "auto",
  "soxr": false,
  "soxrconf": [
    20,
    50,
    91.3,
    100,
    0,
    0
  ],
  "version": "0.23.6-1"
}

In addition BT setting on System page DO seem to survive reboot but backup file I placed in /boot folder of new SD card DIDNT get picked up so no restore of original system settings. I don't know what the sound levels are like as I can't get to a position to check this out.

Chris

**Reply by phreatic62** [reply]
Mine was a fresh build, but my backup was picked up 🙂

**Comment by ceejayemm** [comment]
@rern 

This might be more of a hinderence than a help but...

This morning I applied the UPDATE branch (just because I saw some work had been done in this area and I wondered if any fixes for this issue were in there).  The update applied OK until it got to show:

/srv/http/settings/system-i2s.json
Then a long pause before:
Cannot access /srv/http/bash/cmd.sh : permission denied  (didn't get an exact copy of this error)
Then another long pause before:
chown: cannot access /srv/http/data/mpd/mpd.db : No such directory
Then another long pause before (again):
chown: cannot access /srv/http/data/mpd/mpd.db : No such directory

Then the update finishes.  The errors above are probably correct as I use SharedData on this RPi and the database is actually in:
/srv/http/shareddata

After this update everything seems to work OK except I still get the flashing rAudio-1 logo on the Player page.

As I said  above I hope this is helpful and not just noise getting in the way.

Chris

**Comment by ceejayemm** [comment]
My issue is solved, I don't know if this will be the case for phreatic62 

The clue to solving this was the apparent unavailability of database as noted in my last post.  It turns out that the backup I thought hadn't worked DID in fact work as I should have noticed by most of my settings being available after flashing the new card. The thing that caused me the problems was that the backup restored the Shared Data button position (System page) to ON but in fact the Shared Data path was NOT restored. So it looked like I had Shared Data available when in fact I didn't. Entering the Shared Data path enabled the Player setting page to work correctly.  I have checked this again by re-flashing another card exactly as before using the same backup file and hey presto the Player page flashes the rAudio-1 logo until such time as I enter the Shared Data path.

I still needed to use the command:

sed -i 's|amixer.*2|cat $dirshm/btclient 2> /dev/null|' /srv/http/bash/player-data.sh

to remove the original error in phreatic62 first post. BT settings also survive a reboot.

@rern - I think you need to make sure that the Shared Data path is included in the Backup process run from the System settings page.

@phreatic62 - check your Shared Data path has been restored from the backup - if you indeed use this option.

Chris

**Reply by phreatic62** [reply]
Hi Chris, I'm not at home at the mo, so can't test the volume issue, however I can access it remotely and the network share looks ok, e.g. if I select an album it replaced the radio station that was playing.

Paul

**Reply by ceejayemm** [reply]
Paul

My settings for volume with the SoundCore 2 are: (Player Options page)

Mixer device output Loudspeaker icon - Mixer Device Volume 100%
Bit Perfect No Volume - on
ffmpeg - on

If you can ssh into the RPi you can also look at the 'alsamixer' volume setting and adjust if necessary.

Then I adjust the volume as needed using the SoundCore 2 controls on the top of the device  Sound output volume levels are good for me right now..

Chris

**Reply by phreatic62** [reply]
Hi both.
I don't seem to have all of those options when I look in 'Player' and the speaker icon comes and goes, but can usually be brought back with a refresh.
<img src="https://user-images.githubusercontent.com/61110813/164625815-2c807ddc-4891-4cb3-a52e-c243330b59cc.png" width=300>

I'm not familiar with alsa mixer, buut it seems pretty straitghtforward - if I'm interpreting it correctly :)
Even when sount is coming (quietly) out of my Soundcore, I see the screen below, looking at the website, I should see headphones and soundcore if it's connected. So not sure what's going on there. The volume control for the headphones, as expected, has no effect on the soundcore.
<img src="https://user-images.githubusercontent.com/61110813/164626362-baef39c8-d598-4d41-a0c5-0c55944d3c49.png" width=300>

Bluetooth on is still not surviving a reboot, although when I turn it back on, it seems to remember previous pairings.

Best
Paul

**Reply by ceejayemm** [reply]
Hi Paul

Sorry I misled you yesterday, my settings with the BT speakers connected and active ARE the same as yours. I had been listening yesterday, then stopped to take a phone call and hadn't realised the SoundCore BT connection had timed out just before writing the message above.  

The only other thing I use is on the System / Onboard Wireless page click on the Cog icon next to the Bluetooth option. I have Discoverable by Senders turned ON and Sampling 16bit 44.1kHz to receivers turned OFF. Don't why this would make any difference but its what I have used ever since enabling BT.

Other than this I am now out of ideas to help solve your particular problem.

Chris

**Comment by phreatic62** [comment]
Ok, since today's update, Bluetooth seems to survive a reboot. 🙂
I'm not in a position to try the volume yet, bit I'll report back when we do.

**Comment by phreatic62** [comment]
Hi Both, 
BT is surviving a reboot, however sound is still quiet. 
Alsamixer still only shows the headphone socket and excecuting
"sed -i 's|amixer.*2|cat $dirshm/btclient 2> /dev/null|' /srv/http/bash/player-data.sh" does not work as it can't find player-data.sh, it's actually in /srv/http/bash/sttings/ , if I edit the line to include 'settings', the command seems to run, but nothing happens.

Making a copy of player-data.sh in '/srv/http/bash' also seems ineffective.

regards
Paul
